src/docfx/localization/LocalizationConvention.cs

```diff
@@ -21,38 +21,49 @@ internal static class LocalizationConvention
         /// // TODO: org name can be different
         /// </summary>
         /// <returns>The loc remote url</returns>
-        public static string GetLocalizationRepo(LocalizationMapping localizationMapping, string remote, string locale, string defaultLocale)
+        public static (string remote, string branch) GetLocalizationRepo(LocalizationConfig localizationConfig, string remote, string branch, string locale, string defaultLocale)
         {
-            if (localizationMapping != LocalizationMapping.Repository && localizationMapping != LocalizationMapping.RepositoryAndFolder)
+            if (localizationConfig == null)
             {
-                return remote;
+                return (remote, branch);
+            }
+
+            if (localizationConfig.Mapping != LocalizationMapping.Repository && localizationConfig.Mapping != LocalizationMapping.RepositoryAndFolder)
```

Comment:
make it shorter since it is just folder mapping here?
```csharp
if (localizationConfig.Mapping == LocalizationMapping.Folder)
```

Reply:
i want to use the write list, not black list in case we add more mappings later

---
In reply to: [227691489](https://github.com/dotnet/docfx/pull/3528#discussion_r227691489) [](ancestors = 227691489)